backend/api/Areas/Documents/DocumentController.cs

```diff
@@ -87,11 +87,38 @@ public IActionResult GetRelationshipDocuments(DocumentRelationType relationshipT
         }
 
         /// <summary>
-        /// Deletes the specific document relationship for the given type.
+        /// Uploads a document for the given relationship.
         /// </summary>
         /// <param name="type">Used to identify document type.</param>
         /// <param name="entityId">Used to identify document's parent entity.</param>
         /// <returns></returns>
+        [HttpPost("upload/{relationshipType}/{parentId:long}")]
+        [Produces("application/json")]
+        //[HasPermission(Permissions.PropertyAdd)]
+        [ProducesResponseType(typeof(DocumentUploadResponse), 200)]
+        [SwaggerOperation(Tags = new[] { "documents" })]
+        public async Task<IActionResult> UploadDocumentWithParent(
+            DocumentRelationType relationshipType,
+            long parentId,
+            [FromForm] DocumentUploadRequest uploadRequest
+            )
+        {
+            switch (relationshipType)
+            {
+                case DocumentRelationType.Activities:
+                    var response = await _documentService.UploadActivityDocumentAsync(parentId, uploadRequest);
+                    return new JsonResult(response);
+                default:
+                    throw new BadRequestException("Relationship type not valid.");
+            }
```

Comment:
question - I believe we kinda agreed to do this kind of logic in the service to try to keep controllers as lean as possible

Reply:
Well we chated on it, but did not conclude on where the logic should be. I think we have a meeting later this week to go about that as well as the routes